compiler/rustc_infer/src/infer/mod.rs

```diff
@@ -1184,6 +1211,20 @@ impl<'tcx> InferCtxt<'tcx> {
         }
     }
 
+    pub fn var_for_effect(&self, param: &ty::GenericParamDef) -> GenericArg<'tcx> {
+        let effect_vid = self.inner.borrow_mut().effect_unification_table().new_key(None);
+        // todo maybe we should just use bool here
+        ty::Const::new_infer(
+            self.tcx,
+            ty::InferConst::EffectVar(effect_vid),
+            self.tcx
```

Comment:
Debug assert that the type is always bool probably.

Reply:
done